CMakeLists.txt

```diff
@@ -118,8 +118,12 @@ endforeach()
 set(_SwiftFoundation_wasi_libc_flags)
 if(CMAKE_SYSTEM_NAME STREQUAL "WASI")
     list(APPEND _SwiftFoundation_wasi_libc_flags
+        "SHELL:$<$<COMPILE_LANGUAGE:Swift>:-Xcc -D_WASI_EMULATED_GETPID>"
+        "SHELL:$<$<COMPILE_LANGUAGE:Swift>:-Xfrontend -public-autolink-library -Xfrontend wasi-emulated-getpid>"
```

Comment:
Previously we've used `-public-autolink-library` to have clients of the module link in the listed library which was needed for the static build. In this case, is this to support clients linking as part of the static SDK (if so should we put this in an `if(BUILD_SHARED_LIBS)` path?) or is this just for FoundationEssentials to link these libraries (if so, do we need a public auto link or is it enough to just add the equivalent of `-lwasi-emulated-getpid`?)

Reply:
Good catch! We don't support shared library on WASI for now, but for shared libraries, we should link emulation libraries to shared libraries instead of asking clients to link them.